system/lib/libc/musl/src/internal/pthread_impl.h

```diff
@@ -129,19 +141,28 @@ int __libc_sigprocmask(int, const sigset_t *, sigset_t *);
 void __lock(volatile int *);
 void __unmapself(void *, size_t);
 
-int __timedwait(volatile int *, int, clockid_t, const struct timespec *, void (*)(void *), void *, int);
-void __wait(volatile int *, volatile int *, int, int);
+void __vm_wait(void);
+void __vm_lock(void);
+void __vm_unlock(void);
 
+int __timedwait(volatile int *, int, clockid_t, const struct timespec *, int);
+int __timedwait_cp(volatile int *, int, clockid_t, const struct timespec *, int);
+void __wait(volatile int *, volatile int *, int, int);
+static inline void __wake(volatile void *addr, int cnt, int priv)
+{
+	if (priv) priv = 128;
+	if (cnt<0) cnt = INT_MAX;
 #ifdef __EMSCRIPTEN__
-#define __wake(addr, cnt, priv) emscripten_futex_wake((void*)addr, (cnt)<0?INT_MAX:(cnt))
+	emscripten_futex_wake((void*)addr, (cnt)<0?INT_MAX:(cnt));
```

Comment:
Not sure. I seem to have resolved my issues building the browser tests, and do indeed see lots of breakage. I imagine this is in there.

Re: rationale, https://github.com/bpowers/musl/commit/56fbaa3bbe73f12af2bfbbcf2adb196e6f9fe264 is the musl commit that changed the pthread structs to be volatile.